src/server/pps/server/master.go

```diff
@@ -24,6 +26,19 @@ const (
 	masterLockPath = "_master_lock"
 )
 
+func isFailure(reason string) bool {
+	failReasons := []string{
```

Comment:
this could be simplified as a `map[string]bool` then return `failReasons[reason]`, which returns `false` in the nonexistent case (default value of a bool is false)

https://play.golang.org/p/krYiE4pWwW

Reply:
Good suggestion, implemented.